### PR TITLE
forge: add search method

### PR DESCRIPTION
**Pull request by edvbld**
Hi all,

please review this patch that adds the `Forge.search` method to allow searching for a commit with a particular hash on a forge.

Thanks,
Erik

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Reviewers
 * [Robin Westberg](https://openjdk.java.net/census#rwestberg) (@rwestberg - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/skara pull/854/head:pull/854`
`$ git checkout pull/854`

### CODE REVIEW

File: forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabHost.java [hunk 1]
```diff
@@ -210,4 +216,44 @@ public boolean isMemberOf(String groupId, HostUser user) {
...
+    public Optional<HostedCommitMetadata> search(Hash hash) {
+        var hex = hash.hex();
+        for (var group : groups) {
+            var projects = request.get("groups/" + group + "/projects")
+                                  .execute()
+                                  .stream()
+                                  .map(o -> o.get("id").asString())
+                                  .collect(Collectors.toList());
+            for (var project : projects) {
+                var c = request.get("projects/" + project + "/repository/commits/" + hex)
```
Comment (rwestberg) [inline forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabHost.java:246]: Could we use this endpoint instead? https://docs.gitlab.com/ce/api/commits.html#get-a-single-commit

**Review by rwestberg — COMMENTED** [review]
Looks good, a bit worried that auto-pagination of thousands of commits will take a while though.. :)

File: forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabHost.java (hunk 1 above)
Comment (edvbld) [reply forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabHost.java:246]: That is the endpoint the code is using :smile_cat: Did you perhaps miss the `+ hex` part?

**Comment by edvbld** [comment]
> Looks good, a bit worried that auto-pagination of thousands of commits will take a while though.. :)

Since we are limiting the GitHub search to selected organizations, we should for the https://github.com/openjdk organization get at most `O(#repos)` results.

File: forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabHost.java (hunk 1 above)
Comment (rwestberg) [reply forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabHost.java:246]: Well then, carry on!

**Review by rwestberg — APPROVED** [review]

**Pull request closed without merging**